File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    
    public static int gcd(int a, int b){
            while(b != 0) {
                int r = a % b;
                a = b;
                b = r;
            }
            return a;
        }
    
    public int[] solution(int denum1, int num1, int denum2, int num2) {
        int[] answer = new int[2];
        int num = denum1 * num2 + denum2 * num1;
        int den = num1 * num2;
        int result = Solution.gcd(num, den);
        answer[0] = num/result;
        answer[1] = den/result;
        return answer;
    }
}

File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.js
// gcd : 최대공약수
function solution(denum1, num1, denum2, num2) {
    let ans = [];
    let num = denum1 * num2 + denum2 * num1;
    let den = num1 * num2;
    let gcd = (a, b) => {
        return (a % b === 0 ? b : gcd(b, a % b));
    }
    let y = gcd(num, den);
    ans[0] = num / y;
    ans[1] = den / y;
    return ans;
     
}